packages/calcite-components/src/utils/focusTrapComponent.ts

```diff
@@ -58,9 +71,10 @@ export function connectFocusTrap(component: FocusTrapComponent, options?: Connec
   }
 
   const focusTrapOptions: FocusTrapOptions = {
-    clickOutsideDeactivates: true,
-    escapeDeactivates: false,
+    clickOutsideDeactivates: component.clickOutsideDeactivates ?? true,
```

Comment:
This setup seems a bit confusing to me. We’re passing options when setting up the focus trap, but some are also defined on the component itself. This could could cause confusion or issues (e.g., `onDeactivate` being set in both `component.onFocusTrapDeactivate` and `options.onFocusTrapDeactivate`). Can we roll it back to just rely on the component passing options for now? I'd like us to revisit this approach later.